source/Ocl/Parsing/OclParser.cs

```diff
@@ -57,16 +57,22 @@ public static Parser<T> SameLineToken<T>(this Parser<T> parser)
             from close in ArrayClose.Token()
             select values.ToArray();
 
-        private static readonly Parser<object> Literal =
-            QuotedStringParser.QuotedStringLiteral
-                .Or<object>(HeredocParser.Literal)
-                .Or(DecimalLiteral.Select(d => (object)d))
-                .Or(IntegerLiteral.Select(d => (object)d))
-                .Or(EmptyArrayLiteral)
+        static readonly Parser<object> ArrayLiteral =
```

Comment:
The key difference between `Or` and `XOr` in Sprache is around backtracking on partial failed matches. `XOr` wouldn't have worked here because all these parsers start with `[`, so `XOr` would try the first parser and if it partially matched and then failed, wouldn't even try the others. 

You could restructure the parsers so that they work with `XOr`, but without testing, not 100% sure if that'll bring us any benefits.

Reply:
Ah ofcourse, it's the 2nd character that is the differentiator, so we could restructure it but it'd make it less clear.